airflow/settings.py

```diff
@@ -278,17 +278,24 @@ def remove(*args, **kwargs):
         pass
 
 
+def _is_sqlite_db_path_relative(sqla_conn_str: str) -> bool:
+    """Determine whether the database connection URI specifies a relative path."""
+    import re2
+
+    return (
+        bool(sqla_conn_str)
+        and sqla_conn_str.startswith("sqlite")
+        # In memory is not useful for production, but useful for writing tests against Airflow for extensions
+        and sqla_conn_str != "sqlite://"
+        and not os.path.isabs(re2.sub(r"^sqlite:///", "", sqla_conn_str))
+    )
```

Comment:
If this is to be split into a separate function, I would use early returns to make this giant `and` chain more readable. Also I would not use regex here; it is enough to simply check there are exactly three slashes, and feed `sqla_conn_str[10:]` to `isabs`.

Reply:
I don't like magic numbers, would it be acceptable to use
```python
    if (
        sqla_conn_str.startswith(abs_prefix := "sqlite:///")
        and os.path.isabs(sqla_conn_str[len(abs_prefix):])
    ):
```
?